### PR TITLE
Introduce new type dev_optional

### PR DESCRIPTION
**Pull request by MarshallOfSound**
NOTE: This is a breaking change, Galactus doesn't know about `DEV_OPTIONAL`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
     "@types/yargs": "^8.0.1",
     "chai": "^4.1.2",
     "mocha": "^4.1.0",
+    "sane": "^2.5.0",
```
Comment (malept) [inline package.json:26]: Is this used anywhere? The tests (for the `fsevents` test?)

**Review by malept — CHANGES_REQUESTED** [review]
The new depType should be documented in the README.

Otherwise, :+1:

File: package.json (hunk 1 above)
Comment (MarshallOfSound) [reply package.json:26]: Yeah, used for the `fsevents` test 👍

File: test/Walker_spec.ts [hunk 2]
```diff
@@ -33,4 +33,11 @@ describe('Walker', () => {
   it('should locate a dep of a dev dep that is also a top level prod dep as a prod dep', () => {
     expect(dep('debug')).to.have.property('depType', DepType.PROD);
   });
+
+  it('should locate a dep of a dev dep that is optional as a dev_optional dep', function () {
```
Comment (malept) [inline test/Walker_spec.ts:37]: arrow function like the rest?

Comment (MarshallOfSound) [reply test/Walker_spec.ts:37]: `this.skip` only works on non-arrow functions (needs to be bound correctly)

**Comment by malept** [comment]
In `README.md`:

```js
// DepType.PROD --> Production dependency
// DepType.OPTIONAL --> Optional dependency
// DepType.DEV --> Development dependency
// DepType.DEV_OPTIONAL --> Optional dependency of a development dependency
// DepType.ROOT --> The root module
```

**Comment by MarshallOfSound** [comment]
Adds docs, will major bump and merge on green build

**Pull request merged**